### PR TITLE
Feature/new line chart

### PR DESCRIPTION
**Pull request by noahmanger**
Implements the new chart for showing cumulative amount raised and spent. 
- Creates a new `LineChart` component that builds a chart with a line for each entity type
- The cursor line moves with the mouse on hover and can be controlled via the arrow buttons
- Moving the cursor changes the totals to the right
- The totals represent the amount of money for the ending coverage date *up to* the first of the month (excluding the money reported *on* the first)

demo

## Some notes:
After I got data from @PaulClark2 , I learned that given the very different rules that apply to all the committees and the irregular primary schedule, there's actually reports for all committees in just about every month, which allows us to show data points for every month. But the amount reported on any given date varies wildly depending on how prevalent of a filing deadline it is; this would be very confusing if we were showing the amount reported *in* that period, because some would be drastically different than others. But because we're showing the *cumulative* amount raised until that point, I think it works well.

I opted to have the cursor move on hover largely because it was the simplest solution, but if we want to have it move on click or drag, we can do that. As is, on mobile the cursor moves on tap (it should eventually be on drag).

I also haven't gotten to re-implementing everything else around this — the new methodology buttons and the accordions. I think it might make sense to follow up with that separately next sprint.

## Remaining to-do:
- [ ] Add "other spenders" back in
- [ ] Add spending data
- [ ] Clean up CSS and JS

Requires https://github.com/18F/fec-style/pull/460
Resolves https://github.com/18F/openFEC-web-app/issues/1361

### CODE REVIEW

File: static/js/modules/line-chart.js [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+
+  var minYear = d3.min(data, function(d) {
+    return d.date.getFullYear();
+  });
+
+  var maxYear = minYear + 1;
+
+  // Build the scales
+  var x = d3.time.scale()
+    // .domain(d3.extent(data, function(d) {
```
Comment (xtine) [inline static/js/modules/line-chart.js]: what did this do?

File: static/js/modules/line-chart.js [hunk 2]
```diff
@@ -0,0 +1,243 @@
...
+        return parseMY(d);
+      } else if (d.getMonth() % 4 === 0) {
+        return parseM(d);
+      } else {
+        return '';
+      }
+    };
+  }
+
+    axis.ticks(d3.time.month)
```
Comment (xtine) [inline static/js/modules/line-chart.js]: spacing

**Comment by xtine** [comment]
When I click on "Methodology," the modal appears but the dimensions are bigger than my screen:
screen shot 2016-08-11 at 9 00 20 pm

File: static/js/modules/line-chart.js (hunk 1 above)
Comment (noahmanger) [inline static/js/modules/line-chart.js]: Oops, I meant to remove that. It was defining the min and max of the x-scale by the min and max dates in the data set, but I think it's better to show the full two years. I'll remove it.

File: static/js/modules/line-chart.js [hunk 3]
```diff
@@ -0,0 +1,240 @@
...
+        date: d.date,
+        amount: d[type]
+      };
+    });
+
+    var entityMax = d3.max(totals, function(d) {
+      return d.amount;
+    });
+
+    max = entityMax > max ? entityMax : max;
```
Comment (adborden) [inline static/js/modules/line-chart.js]: Can you use `Math.max` here?

Comment (noahmanger) [inline static/js/modules/line-chart.js]: Bloops. Yeah. 😎

**Comment by xtine** [comment]
This is so great. One suggestion though: is it possible to set `overflow: hidden` on body when the modal is triggered? Modals look great on my screen but when I try to scroll the content of the modal the body underneath also scrolls.

**Comment by adborden** [comment]
Oops, sorry, I had an old page open, just missed @xtine's comment.

**Comment by noahmanger** [comment]
Oh oops. I just pushed a commit with `Math.max` and disabling scrolling on `body` when it's opened. I need to make a new PR into feature anyway.

**Pull request merged**